### PR TITLE
BBS+ Signature and Proof Verification

### PR DESCRIPTION
**Pull request by man2706kum**
This PR includes the solidity smart contract implementation of the BBS+ signature verification and proof verification along with tests. 
The following functions are implemented:
- `proofVerify`: the BBS+ proof verification function
- `verifySignature`: the BBS+ signature verification function
- all the related utils functions

### CODE REVIEW

File: src/bbs_verify.sol [hunk 1]
```diff
@@ -0,0 +1,705 @@
...
+
+        for (uint256 i = 0; i < (input.length + 1) / 2; i++) {
+            output[i] = input[input.length - 1 - i];
+            output[input.length - 1 - i] = input[i];
+        }
+
+        return output;
+    }
+
+    function proofVerify(
```
Comment (Mikerah) [inline src/bbs_verify.sol]: Keep the naming consistent. So, this should be `verifyProof`

File: src/bbs_verify.sol [hunk 2]
```diff
@@ -0,0 +1,705 @@
...
+            if (!isPresent[i]) {
+                complementSet[index] = i;
+                index++;
+            }
+        }
+
+        return complementSet;
+    }
+
+    function proofChallengeCalculate(
```
Comment (Mikerah) [inline src/bbs_verify.sol]: Same here

File: src/bbs_verify.sol (hunk 1 above)
Comment (man2706kum) [reply src/bbs_verify.sol]: fixed

File: src/bbs_verify.sol (hunk 2 above)
Comment (man2706kum) [reply src/bbs_verify.sol]: fixed

**Review by Mikerah — APPROVED** [review]

**Pull request merged**